cobalt/android: Reduce GPU mem & image decode cache size

Awallky opened the pull request:
Currently when browsing on the home screen, watching a video, and returning back to browsing on the home screen, there is a significant spike in malloc allocations from before video playback. This has been observed to be 3x increase in malloc. In an attempt to mitigate some of this, Cobalt reduces the maximum amount of GPU allocations and image decode cache that can be made.

This reduces the LimitImageDecodeCacheSize to 24 MB. It also reduces the total GPU memory size to 32 MB as even chrome's current low end device limits are too high for Cobalt's needs. Finally, this fixes a test error introduced in https://lbshell-internal-review.git.corp.google.com/c/cobalt_src/+/307358.

Some effects of this change are seen in the test results below, the overall PSS and malloc levels

`PSS` & `malloc` before change (respectively):
  - splash screen - 260.9 MB, 56 MB
  - video playback - 470 MB, 77.8 MB
  - splash screen -  422 MB, 173.5 MB
`PSS` & `malloc` after change (respectively):
  - splash screen - 268 MB, 56 MB
  - video playback - 405 MB, 80 MB
  - splash screen - 389 MB, 147.2 MB

Bug: 405467220

yell0wd0g reviewed (approved):
Changes LGTM. It'd be better to prefix to the title
an indication of where the delta is, e.g.
`cobalt/android: Reduce GPU mem & image decode cache size`.

Did you see a measurable difference in the CC category
mem use before/after this change? On any scenario of
your choice, since at this point we haven't found an
"official" way of measuring. If so then it'd be great
to just leave a word about it in the CL/PR description.

Awallky commented:
> Did you see a measurable difference in the CC category
mem use before/after this change? On any scenario of
your choice, since at this point we haven't found an
"official" way of measuring. If so then it'd be great
to just leave a word about it in the CL/PR description.

I went ahead and modified the description detailing what I observed in the before/after test runs.

Also, modified the title per your suggestion.
